app/views/employment/CheckEmploymentDetailsView.scala.html

```diff
@@ -57,10 +57,10 @@
         val payeRef: (String, Option[String]) = ("payeRef", employerRefData)
         val directorCeasedDate :(String, Option[String]) = ("directorCeasedDate", employmentDetails.flatMap(_.directorshipCeasedDate.flatMap(dateFormatter)))
         val closeCompany: (String, Option[String]) = ("closeCompany", convertBoolToYesOrNo(employmentDetails.flatMap(_.closeCompany)))
-        val payReceived: (String, Option[String]) = ("payReceived", employmentDetails.map(e => bigDecimalCurrency(e.pay.taxablePayToDate.toString)))
-        val totalTaxToDate: (String, Option[String]) = ("totalTaxToDate", employmentDetails.map(e => bigDecimalCurrency(e.pay.totalTaxToDate.toString)))
+        val payReceived: (String, Option[String]) = ("payReceived", employmentDetails.map(e => bigDecimalCurrency(e.pay.get.taxablePayToDate.get.toString)))
+        val totalTaxToDate: (String, Option[String]) = ("totalTaxToDate", employmentDetails.map(e => bigDecimalCurrency(e.pay.get.totalTaxToDate.get.toString)))
         val tipsAndOtherPaymentsAmount: (String, Option[String]) = (s"tipsAndOtherPayments.${if(user.isAgent) "agent" else "individual"}",
-            employmentDetails.flatMap(_.pay.tipsAndOtherPayments.map(amount => bigDecimalCurrency(amount.toString))))
+            employmentDetails.flatMap(_.pay.get.tipsAndOtherPayments.map(amount => bigDecimalCurrency(amount.toString))))
```

Comment:
these gets aren't safe - as in if it is empty you will get a NullPointerException as it's trying to get something when there's nothing there.

Instead you should use flatMaps to see if the value is there before handing it to the bigDecimalCurrency method. See line 58 in here